storage/rocksdb/ha_rocksdb.cc

```diff
@@ -7841,6 +8001,20 @@ int ha_rocksdb::check_uniqueness_and_lock(
       return rc;
     }
 
+    /*
+      If the pk key has ttl, we may need to pretend the row wasn't
+      found if it is already expired. The pk record is read into
+      m_retrieved_record by check_and_lock_unique_pk().
+    */
+    if (is_pk(key_id, table, m_tbl_def) && found && m_pk_descr->has_ttl() &&
+        should_hide_ttl_rec(rocksdb::Slice(&m_retrieved_record.front(),
+                                           m_retrieved_record.size()),
+                            (row_info.tx->m_snapshot_timestamp
+                                 ? row_info.tx->m_snapshot_timestamp
```

Comment:
This seems to be the only place you check for m_snapshot_timestamp being 0, but all of the other ones assume it's not. All of the other callsites always have a snapshot involved?